jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -2453,22 +2447,59 @@ ecma_string_trim_helper (const lit_utf8_byte_t **utf8_str_p, /**< [in, out] curr
     current_p += read_size;
   }
 
-  current_p = *utf8_str_p + *utf8_str_size;
+  *utf8_str_size = (lit_utf8_size_t) (*utf8_str_p + *utf8_str_size - nonws_start_p);
+  *utf8_str_p = nonws_start_p;
+} /* ecma_string_trim_front */
 
-  while (current_p > nonws_start_p)
-  {
-    read_size = lit_read_prev_code_unit_from_utf8 (current_p, &ch);
+/**
+ * Helper function for trimming.
+ *
+ * Used by:
+ *        - ecma_string_trim_helper
+ */
+void ecma_string_trim_back (const lit_utf8_byte_t **utf8_str_p, /**< [in, out] current string position */
+                             lit_utf8_size_t *utf8_str_size)  /**< [in, out] size of the given string */
```

Comment:
I suggest to change the prototype of the new functions to be
```c
const lit_utf8_byte_t *
ecma_string_trim_front (const lit_utf8_byte_t *start_p, const lit_utf8_byte_t *end_p);
const lit_utf8_byte_t *
ecma_string_trim_back (const lit_utf8_byte_t *start_p, const lit_utf8_byte_t *end_p);
```
where the return value is the new start/end position respectively.